### PR TITLE
added Nutanix platform to openshift/api

### PR DESCRIPTION
**Pull request by yannickstruyf3**
This PR adds the Nutanix platform to the openshift/api repo.

### CODE REVIEW

File: config/v1/types_infrastructure.go [hunk 1]
```diff
@@ -169,6 +169,9 @@ const (
 
 	// AlibabaCloudPlatformType represents Alibaba Cloud infrastructure.
 	AlibabaCloudPlatformType PlatformType = "AlibabaCloud"
+
+	// NutanixPlatformType represents Nutanix infrastructure.
+	NutanixPlatformType PlatformType = "Nutanix"
```
Comment (staebler) [inline config/v1/types_infrastructure.go:174]: This value needs to be added to the `+kubebuilder:validation:Enum` annotation on the `PlatformType` type.

Comment (yannickstruyf3) [reply config/v1/types_infrastructure.go:174]: @staebler  Added the value as requested.

**Comment by staebler** [comment]
@sttts These changes look good to me. The Nutanix platform is coming in 4.11. Would you prefer holding off on merging this until we open for 4.11? Or is it OK to get this in now to make it easier for putting together and testing other work that depends on these changes?

File: config/v1/types_infrastructure.go [hunk 2]
```diff
@@ -678,6 +689,23 @@ type AlibabaCloudResourceTag struct {
...
+// This only includes fields that can be modified in the cluster.
+type NutanixPlatformSpec struct{}
+
+// NutanixPlatformStatus holds the current status of the Nutanix infrastructure provider.
+type NutanixPlatformStatus struct {
+	// apiServerInternalIP is an IP address to contact the Kubernetes API server that can be used
+	// by components inside the cluster, like kubelets using the infrastructure rather
+	// than Kubernetes networking. It is the IP that the Infrastructure.status.apiServerInternalURI
+	// points to. It is the IP for a self-hosted load balancer in front of the API servers.
+	APIServerInternalIP string `json:"apiServerInternalIP,omitempty"`
```
Comment (sttts) [inline config/v1/types_infrastructure.go:702]: why do we need this in addition?

Comment (thunderboltsid) [reply config/v1/types_infrastructure.go:702]: @sttts AFAIK on-prem platforms are expected to provide both these fields to the machine config operator. see: https://github.com/openshift/machine-config-operator/pull/2852/files#

**Comment by tkashem** [comment]
LGTM

/assign @benluddy @p0lyn0mial

**Comment by sdodson** [comment]
Adding `bugzilla/valid-bug` label while we sort out whether or not this repo is misconfigured.

**Comment by tkashem** [comment]
retitle Bug 2049838: added Nutanix platform to openshift/api

**Comment by tkashem** [comment]
/retitle Bug 2049838: added Nutanix platform to openshift/api

**Pull request merged**